documentation/developer/adr/0005-record-cache-vs-recreate.md

```diff
@@ -0,0 +1,49 @@
+# 1. Record architecture decisions
+
+Date: 2022-06-15 - 2022-11-30
+
+## Status
+
+Proposed
+
+## Context
+
+At present we recreate FreeCAD geometries by travelling through the boundaries of their
+"Matryoshka". We have a number of issues relating to this as our implementation apppears
+ to be imperfect.
+
+Assumption: `BluemiraGeo` objects are "static".
+
+Options:
+* Continue with recreating the shapes
+    * Pros:
+        * We can track labels throughout the creation of geometry
+        * Meshing and FE problems are easier to set up because of the above
+    * Cons:
+        * Higher computational cost (recreate upon each call of geometry)
+        * Higher maintenance overhead
+        * Reverse engineering of FreeCAD is presently imperfect
+            * An operation can work in FreeCAD but fail upon "reconstruction" in Bluemira
+            * NotClosedWireError, DisjointedFaceError, ...
+* Cache the FreeCAD shape
+    * Pros:
+        * Lower computational cost
+        * Lower maintenance cost
+        * Less logic and code
+    * Cons:
+        * If we want to label specific boundaries of geometry, we need to travel through
+        all the boundaries and find the "right" ones and label post-instantiation (fiddly!)
+        * This means that we cannot track shared boundaries (upon creation, but see above)
+        * Meshing and FE problems are more complicated to set up
+
+## Decision
+
+The recreation of the shapes was causing serious, frequent, often not-perfectly-reproducible
+problems when trying to run the code. After experimenting with geometry caching, noting that
+it was more stable and faster, we decided to cache the underlying FreeCAD shapes, rather than
+recreate them.
+
```

Comment:
Maybe a comment here that it may be possible to revisit this in future with FreeCAD V1 when its released